control/trajectory_follower_nodes/src/controller_node.cpp

```diff
@@ -0,0 +1,172 @@
+{
+  using std::placeholders::_1;
+
+  const double ctrl_period = declare_parameter<double>("ctrl_period", 0.03);
+  timeout_thr_sec_ = declare_parameter<double>("timeout_thr_sec", 0.5);
+
+  const auto lateral_controller_mode =
+    getLateralControllerMode(declare_parameter("lateral_controller_mode", "mpc_follower"));
+  switch (lateral_controller_mode) {
+    case LateralControllerMode::MPC: {
+      lateral_controller_ = std::make_shared<trajectory_follower::MpcLateralController>(*this);
+      break;
+    }
+    case LateralControllerMode::PURE_PURSUIT: {
+      lateral_controller_ = std::make_shared<pure_pursuit::PurePursuitLateralController>(*this);
+      break;
+    }
+    default:
+      throw std::domain_error("[LateralController] invalid algorithm");
+  }
+
+  const auto longitudinal_controller_mode =
+    getLongitudinalControllerMode(declare_parameter("longitudinal_controller_mode", "pid"));
+  switch (longitudinal_controller_mode) {
+    case LongitudinalControllerMode::PID: {
+      longitudinal_controller_ =
+        std::make_shared<trajectory_follower::PidLongitudinalController>(*this);
+      break;
+    }
+    default:
+      throw std::domain_error("[LongitudinalController] invalid algorithm");
+  }
+
+  sub_ref_path_ = create_subscription<autoware_auto_planning_msgs::msg::Trajectory>(
+    "~/input/reference_trajectory", rclcpp::QoS{1}, std::bind(&Controller::onTrajectory, this, _1));
+  sub_steering_ = create_subscription<autoware_auto_vehicle_msgs::msg::SteeringReport>(
+    "~/input/current_steering", rclcpp::QoS{1}, std::bind(&Controller::onSteering, this, _1));
+  sub_odometry_ = create_subscription<nav_msgs::msg::Odometry>(
+    "~/input/current_odometry", rclcpp::QoS{1}, std::bind(&Controller::onOdometry, this, _1));
+  control_cmd_pub_ = create_publisher<autoware_auto_control_msgs::msg::AckermannControlCommand>(
+    "~/output/control_cmd", rclcpp::QoS{1}.transient_local());
+
+  // Timer
+  {
+    const auto period_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(
+      std::chrono::duration<float64_t>(ctrl_period));
+    timer_control_ = rclcpp::create_timer(
+      this, get_clock(), period_ns, std::bind(&Controller::callbackTimerControl, this));
+  }
+}
+
+Controller::LateralControllerMode Controller::getLateralControllerMode(
+  const std::string & controller_mode) const
+{
+  if (controller_mode == "mpc_follower") return LateralControllerMode::MPC;
+  if (controller_mode == "pure_pursuit") return LateralControllerMode::PURE_PURSUIT;
+
+  throw std::domain_error("[LateralController] undesired algorithm is selected.");
+  return LateralControllerMode::INVALID;
+}
+
+Controller::LongitudinalControllerMode Controller::getLongitudinalControllerMode(
+  const std::string & controller_mode) const
+{
+  if (controller_mode == "pid") return LongitudinalControllerMode::PID;
+
+  throw std::domain_error("[LongitudinalController] undesired algorithm is selected.");
```

Comment:
This should be removed.

Reply:
remove in https://github.com/autowarefoundation/autoware.universe/pull/901/commits/2d595771f5f018b62eca310dc57351dcd85bb7de